src/main/java/mods/eln/sixnode/tutorialsign/TutorialSignElement.java

```diff
@@ -53,51 +54,54 @@ public static String getText(String balise) {
 				int i = 0;
 			} catch (Exception e) {
 			}
-*/
+            */
             //optional, but recommended
             //read this - http://stackoverflow.com/questions/13786607/normalization-in-dom-parsing-with-java-how-does-it-work
             //doc.getDocumentElement().normalize();
-
-            try {
-                String file = Utils.readMapFile("EA/tutorialSign.txt");
-                String ret;
-                if (file.contains("\r\n"))
-                    ret = "\r\n";
-                else
-                    ret = "\n";
-
-                file = file.replaceAll("#" + ret, "#");
-                file = file.replaceAll(ret + "#", "#");
-
-                String[] split = file.split("#");
-
-                boolean first = true;
-                int counter = 0;
-                String baliseTag = "";
-
-                for (String str : split) {
-                    if (first) {
-                        first = false;
-                        continue;
-                    }
-                    if (counter == 0) {
-                        baliseTag = str;
-                    }
-                    if (counter == 1) {
-                        baliseMap.put(baliseTag, str);
-                    }
-
-                    counter = (counter + 1) & 1;
-                }
-            } catch (IOException e) {
-                //	e.printStackTrace();
-            }
+            String localizedPath = "EA/tutorialSign/" + getCurrentLanguage() + ".txt"; //localized file path
+            String defaultPath = "EA/tutorialSign/en_US.txt"; //if the localized file is not found, then load the default file (en_US)
+            String oldPath = "EA/tutorialSign.txt"; //for compatibility
+            if(Utils.mapFileExists(localizedPath))
+                loadMapFromPath(localizedPath);
+            else if(Utils.mapFileExists(defaultPath))
+                loadMapFromPath(defaultPath);
+            else
+                loadMapFromPath(oldPath);
         }
         String text = baliseMap.get(balise);
         if (text == null) return tr("No text associated to this beacon");
         return text;
     }
 
+    private static void loadMapFromPath(String path) {
+        try {
+            String file = Utils.readMapFile(path);
+            String ret;
+            if (file.contains("\r\n"))
+                ret = "\r\n";
+            else
+                ret = "\n";
+
+            file = file.replaceAll("#" + ret, "#");
+            file = file.replaceAll(ret + "#", "#");
+
+            String[] split = file.split("#");
+
+            boolean add = false;
+            String baliseTag = "";
+
+            for (String str : split) {
+                if (add)
+                    baliseMap.put(baliseTag, str);
+                else
+                    baliseTag = str;
+                add = !add;
+            }
+        } catch (IOException e) {
+            //	e.printStackTrace();
```

Comment:
Place `Utils.println("<some error>");` code here for debugging purposes when an exception is thrown.